plone/app/users/browser/register.py

```diff
@@ -60,9 +72,9 @@ class BaseRegistrationForm(AutoExtensibleForm, form.Form):
     # this attribute indicates if user was successfully registered
     _finishedRegister = False
 
-    def __init__(self, *args, **kwargs):
-        super(BaseRegistrationForm, self).__init__(*args, **kwargs)
-        self.schema = getUtility(IRegisterSchemaProvider).getSchema()
+    @property
```

Comment:
By setting an attribute in the `__init__` method, the code was executed only once. Using a property here is not efficient, because the same code will be executed everytime we access the schema. That's what caused the memory leak just before we merged few days ago.